### PR TITLE
docs: Add guide for sensorless homing

### PR DESCRIPTION
**Pull request by yschroeder**
As discussed in #997 this is my take on the documentation of sensorless homing.

Please read it and give feedback if anything remains unclear or if I missed to cover something.

### CODE REVIEW

**Comment by KevinOConnor** [comment]
Thanks!  I'll look to commit after v0.7.0 release (aiming for the 20th).

One thing I'm not sure about is the filename - I've been working on the mechaduino hardware recently and it too could support sensorless homing - but it would have a totally different configuration.  How about "Trinamic_Tips.md" or something like that?

Also, I noticed a few minor things (none super important):
- the guide says that "Sensorless homing works best at low motor speeds.", but my understanding is that a speed too low may not work.  That is, there is a minimum speed required for the functionality to be stable.
- it would be good to recommend setting "homing_retract_dist=0" to avoid the "second bump" during homing.  Since sensorless homing isn't terribly accurate anyway, there is no gain in the second home, and it can complicate selection of good SGT parameters.
- I saw a misspelling - "acual" instead of "actual".

-Kevin

**Comment by dushyantahuja** [comment]
Would it make sense to move all documentation into the wiki instead of
multiple config/doc files. I can help in transfer of the information.

**Comment by yschroeder** [comment]
Thanks for the feedback, I will update the PR accordingly. However, I am not sure if `Trinamics_Tips.md` goes into the right direction. Because it is not general Tips on Trinamics drivers but solely the homing feature (and only the one of the TMC2130). But maybe there wil be more tips on Trinamics drivers at some point (dream big).

I don't think putting the docs to the wiki is a good idea. I feel that the wiki in GitHub projects is not very visible and tends to rot. Having the docs with the code also ensures that they are always referring to the current version of the code.

**Comment by FHeilmann** [comment]
A small comment, sensorless homing usually requires a certain minimum motor speed to work correctly so you might want to mention that.  I don't have a good source for this unfortunately, maybe there is something in the datasheets.

**Comment by dobryj** [comment]
This documentation looks promising. It is easy to understand this text.
I have only one note. It need to add more information about sensorless homing in non-cartesian kinematics cases. For example CoreXY or delta.

**Comment by jakep82** [comment]
Nothing is different between corexy and cartesian for sensorless homing.

**Comment by cfelicio** [comment]
Sorry, but how is it not different, considering that both motors will stall when homing a corexy printer?

Also, how do we wire the DIAG on RAMPS for Corexy? I saw some documentation on marlin, where they supposedly just pick one (let's say X) and wire it to both X and Y endstops...

**Comment by jakep82** [comment]
I've used sensorless homing on my corexy.  My configuration is exactly as described in the document.  I have one driver defined as X, and the diag pin from that driver is connected to the X endstop pin.  Same for Y.  I believe Klipper looks for a signal on the stepper for each axis, so you can't share an endstop between X and Y.

**Comment by yschroeder** [comment]
I don't have any experience on homing CoreXY or Delta printer. I have only worked with cartesian onces (and my cable winch device 😃).

I believe you would always home all axes on a Delta at once and stop each axis separately when it hits its end stop. Can you even do a `G28 X` on a Delta? It can only home all axes at once, right?

I have no clue for CoreXY how homing works as you need to turn both steppers to home a single axis right? What happens if you hit the mechanical limit? Do both steppers stall?

**Comment by cfelicio** [comment]
I will setup the TMC2130 over the Christmas break on my CoreXY and report back. I will start with the suggestions made by jakep82, and wire one motor per endstop.

**Comment by wizhippo** [comment]
I may have lucked out but on my corexy I have sensorless homing working with my tmc drivers.  Just had to play with the stall sensitivity to find the correct settings. Just plugged in the x and y endstops into the correct pins on the driver boards. I can’t attest for repeatability though as I have never had the need to home and then continue printing.

**Comment by KevinOConnor** [comment]
I'm fine with merging this as is (additions can always be done with later patches).  Let me know if you'd prefer that, or if you prefer waiting for any changes you may have pending.

-Kevin

**Comment by yschroeder** [comment]
I will make some changes in the coming days and update this PR. Please merge afterwards.

**Comment by speendo** [comment]
@KevinOConnor 
You mentioned that
> it would be good to recommend setting "homing_retract_dist=0" to avoid the "second bump" during homing.  Since sensorless homing isn't terribly accurate anyway, there is no gain in the second home, and it can complicate selection of good SGT parameters.

When I setup sensorless homing on my (cartesian) printer, I first tried to set this `homing_retract_dist=0`. However, as far as I remember (cannot check right now) the effect of this setting was, that the carriage didn't move away from the frame after first bumping into it.

This was undesireable, because when I issued a second homing command right after the first one (without moving the axis), the motors would try to turn at quite a high speed although they were homed already resulting in this ugly sound that makes you think you are breaking your printer. Of course, after a second or two the drivers would recognize that they were homed already. Still something you would try to avoid on a regular basis.

When setting `homing_retract_dist=10` or another positive value reasonably far away from zero, The carriage would first home, then move away from the frame and stop there. There is no second bump.

I don't know if this is intended (actually when I have a `G28 X0 Y0` command at the end of a print, this also fails, which might be an indicator that also in the case of `homing_retract_dist` this is some kind of error), but actually this is better than leaving the carriage touching the frame.

File: docs/Sensorless_Homing.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+The name of the virtual end stop pin is derived from the name of the TMC2130 section.
+
+ATTENTION: This guide only mentions the mandatory parameters and the ones needed to set up sensorless homing. There are many other options to configure on a TMC2130, make sure to take a look at `config/example-extras.cfg` for all the available options.
+
+## Testing of SPI communication
+Now that the stepper driver is configured, let's make sure that Klipper can communicate with the TMC2130 by sending the following extended G-Code command to the printer:
+
+```
+TMC_DUMP stepper=stepper_x
```
Comment (trevjonez) [inline docs/Sensorless_Homing.md]:
```suggestion
DUMP_TMC STEPPER=stepper_x
```

Comment (yschroeder) [reply docs/Sensorless_Homing.md]: Thanks for finding this!

**Comment by yschroeder** [comment]
@KevinOConnor: I fixed the remaining issues. Feel free to merge this now.

Is two commits fine or do you prefer this squashed?

**Comment by yschroeder** [comment]
After seeing #1071 I added more info on homing CoreXY machines.

**Comment by KevinOConnor** [comment]
Thanks.

FYI, if a homing_retract_dist causes the head to backoff and not move towards the frame a second time, then something is buggy.  I would not recommend depending on that.

-Kevin

**Comment by jakep82** [comment]
I would need to test further, but I'm pretty sure if it doesn't move back toward the frame, it's because it has triggered during the direction change at the end of the retract.  That means it's now establishing the endstop location based on the retract distance, not on where it contacted the frame.  I still think it's a good idea to recommend using a retract_dist of 0 with sensorless homing.

**Pull request merged**